source/slang/slang-parser.cpp

```diff
@@ -9388,7 +9388,7 @@ static NodeBase* parseBuiltinRequirementModifier(Parser* parser, void* /*userDat
     return modifier;
 }
 
-static NodeBase* parseMagicTypeModifier(Parser* parser, void* /*userData*/)
+static NodeBase* parseMagicTypeModifier2(Parser* parser, void* /*userData*/, bool magicEnum)
```

Comment:
`parseMagicTypeModifierImpl`.

Let's not use bool parameters.

`bool magicEnum` ==> `MagicTypeModifierKind kind`.